File: leetcode/2medium/권민서/1432.java
import java.util.*;

class Solution {
    public int maxDiff(int num) {
        String numStr = String.valueOf(num);
        String maxStr = getMax(numStr);
        int maxVal = Integer.parseInt(maxStr);

        String minStr = getMin(numStr);
        int minVal = Integer.parseInt(minStr);

        return maxVal - minVal;
    }

    private String getMax(String numStr) {
        char target = '\0';
        for (char c : numStr.toCharArray()) {
            if (c != '9') {
                target = c;
                break;
            }
        }

        if (target == '\0') return numStr;

        return numStr.replace(target, '9');
    }

    private String getMin(String numStr) {
        char[] numArr = numStr.toCharArray();

        if (numArr[0] != '1') {
            char target = numArr[0];
            for (int i = 0; i < numArr.length; i++) {
                if (numArr[i] == target) numArr[i] = '1';
            }
            return new String(numArr);
        }

        for (int i = 1; i < numArr.length; i++) {
            if (numArr[i] != '0' && numArr[i] != '1') {
                char target = numArr[i];
                for (int j = i; j < numArr.length; j++) {
                    if (numArr[j] == target) numArr[j] = '0';
                }
                break;
            }
        }
        return new String(numArr);
    }
}

File: leetcode/2medium/오하늘/1432.py
# 에러 나서 찾는중
class Leetcode.이재훈.이재훈.Solution:
    def maxDiff(self, num: int) -> int:
        num_str = str(num)
        
        for digit in num_str: # 최대
            if digit != '9':
                max_num = num_str.replace(digit, '9') # 9로 변경
                break
        else:
            max_num = num_str
        
        if num_str[0] != '1': # 최소 첫번째는 1
            min_num = num_str.replace(num_str[0], '1')
        else:
            for digit in num_str[1:]: 
                if digit != '0' and digit != '1':
                    min_num = num_str.replace(digit, '0')
                    break
            else:
                min_num = num_str

        return int(max_num) - int(min_num)
